subsys/bluetooth/host/adv.c

```diff
@@ -524,34 +535,29 @@ static int hci_set_ad(uint16_t hci_op, const struct bt_ad *ad, size_t ad_len)
 }
 
 static int hci_set_adv_ext_complete(struct bt_le_ext_adv *adv, uint16_t hci_op,
-				    const struct bt_ad *ad, size_t ad_len)
+				    size_t total_data_len, const struct bt_ad *ad, size_t ad_len)
 {
 	struct bt_hci_cp_le_set_ext_adv_data *set_data;
 	struct net_buf *buf;
-	int err;
-	uint8_t max_data_size;
+	const size_t cmd_size = total_data_len +
+				offsetof(struct bt_hci_cp_le_set_ext_adv_data, data);
 
-	buf = bt_hci_cmd_create(hci_op, sizeof(*set_data));
-	if (!buf) {
-		return -ENOBUFS;
+	if (!atomic_test_bit(adv->flags, BT_ADV_EXT_ADV) &&
+	    total_data_len > BT_GAP_ADV_MAX_ADV_DATA_LEN) {
```

Comment:
@Thalley and we already check this before.

So the length of advertising always can place and not need to shorten.